sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -1647,6 +1662,7 @@ class RapidsConf(conf: Map[String, String]) extends Logging {
   }
 
   private def getOptionalCost(key: String) = {
-    conf.get(key).map(toDouble(_, key))
+    // user-provided value takes precedence, then look in defaults map
+    conf.get(key).orElse(optimizerDefaults.get(key)).map(toDouble(_, key))
```

Comment:
Curious, why are these configs being handled with a separate, special-case map rather than as just as other, internal configs?  Treating these like any other RapidsConf config key means we can get nice docstrings for them which could be helpful for generating documentation later if we decide to make them public.

Reply:
My reasoning was that we will at some point have a large number of these default costs that we wouldn't expect users to have to deal with, assuming we can come up with cost models that work well enough in a broad set of use cases. These are quite specialized settings and should probably be a separate section in the documentation, at least.

Perhaps it does make sense to add them as regular configs and we can use a filter to pull these into a separate documentation section when we make them public.

@revans2 Would be good to get your thoughts on this too.

Reply:
I pushed an update to use regular RapidsConf, while still allowing the user to provide costs even if we haven't defined a config entry for a particular operator or expression.